### PR TITLE
add service to backend errors

### PR DESCRIPTION
**Pull request by et**
## Summary

This PR demonstrates populating service name and service version onto error_groups/objects.
I also added `WithServiceName` and `WithServiceVersion` to our go SDK (so one doesn't have to use curl) via options to `Start`. 

## How did you test this change?

https://www.loom.com/share/c73a4c3f58a74fb08a19faa1259329bb

I didn't mention it in the video but I also confirmed that we update open search correctly

Screenshot_2023-07-27_at_4_23_19_PM

## Are there any deployment considerations?

We will hold off on making a new release of the Go SDK. When we are confident this is working correctly, we will create a new git tag for customers to use (tracked in #6149)

### CODE REVIEW

File: e2e/go/fiber.go [hunk 1]
```diff
@@ -15,6 +15,8 @@ import (
 func main() {
 	highlight.SetProjectID("1jdkoe52")
 	highlight.SetOTLPEndpoint("http://localhost:4318")
+	highlight.SetServiceName("go-fiber-app")
+	highlight.SetServiceVersion("abc123")
 	highlight.Start()
```
Comment (et) [inline e2e/go/fiber.go:20]:
```go
highlight.Start(projectId,
  WithServiceName("go-fiber-app"),
  WithServiceVersion("abc123"),
  WithFavoriteColor("blue"),
)
```

File: backend/main.go [hunk 2]
```diff
@@ -235,7 +235,18 @@ func main() {
...
+
+	serviceName := string(runtimeParsed)
+	if runtimeParsed == util.Worker {
+		if handlerFlag != nil {
+			serviceName = *handlerFlag
+		}
+	}
+
+	highlight.Start(
+		highlight.WithServiceName(serviceName),
```
Comment (et) [inline backend/main.go:247]: This is `all` locally
Screenshot 2023-07-28 at 1 11 39 PM

but will change to use the `runtimeFlag` in prod:

Screenshot 2023-07-28 at 1 02 47 PM

File: backend/main.go [hunk 3]
```diff
@@ -235,7 +235,18 @@ func main() {
 			highlight.SetOTLPEndpoint("http://collector:4318")
 		}
 	}
-	highlight.Start()
+
+	serviceName := string(runtimeParsed)
+	if runtimeParsed == util.Worker {
+		if handlerFlag != nil {
+			serviceName = *handlerFlag
```
Comment (et) [inline backend/main.go:242]: Uses the worker flag if this is the worker process.

File: backend/main.go [hunk 4]
```diff
@@ -235,7 +235,18 @@ func main() {
...
+		}
+	}
+
+	highlight.Start(
+		highlight.WithServiceName(serviceName),
+		highlight.WithServiceVersion(os.Getenv("REACT_APP_COMMIT_SHA")),
```
Comment (et) [inline backend/main.go:248]: If this environment variable is not defined (e.g. running locally), this will return an empty string.

**Review by SpennyNDaJets — APPROVED** [review]
Looks good to me! Comments are from our pairing session that I forgot to submit

**Pull request merged**